frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch-utils.ts

```diff
@@ -1,10 +1,26 @@
+import * as _ from 'lodash';
 import { getImpersonate, getActiveCluster } from '../../app/core/reducers';
 import storeHandler from '../../app/storeHandler';
 
 type ConsoleRequestHeaders = {
   'Impersonate-Group'?: string;
   'Impersonate-User'?: string;
   'X-Cluster'?: string;
+  'X-CSRFToken'?: string;
+};
+
+export const getCSRFToken = () => {
+  const cookiePrefix = 'csrf-token=';
+  return (
+    document &&
+    document.cookie &&
+    document.cookie
+      .split(';')
+      .map((c) => _.trim(c))
```

Comment:
I would rather drop the lodash if possible 
```suggestion
      .map((c) => c.trim())
```

Reply:
Updated